[docs][ysql]Added new default roles in ysql. (#10873)
yugabyte/yugabyte-db: docs/content/latest/secure/authorization/rbac-model.md

* Roles inherit the privileges of all other roles granted to them.

View the roles for your YugabyteDB clusters with the following command:

```sql
yugabyte=> \du
```

```output
                                     List of roles
  Role name   |                         Attributes                         | Member of
--------------+------------------------------------------------------------+-----------
 postgres     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
 yb_extension | Cannot login                                               | {}
 yb_fdw       | Cannot login                                               | {}
 yugabyte     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
```

The following table describes the default YSQL roles and users in YugabyteDB clusters.

> | Role | Description |
> | --- | --- |
> | postgres | Superuser role used during database creation and for PostgreSQL operations. |
> | yb_extension | Role that allows non-Superuser users to create PostgreSQL extensions. |
> | yb_fdw | Role that allows non-Superuser users to [CREATE](https://www.postgresql.org/docs/11/sql-createforeigndatawrapper.html), [ALTER](https://www.postgresql.org/docs/11/sql-alterforeigndatawrapper.html), and [DROP](https://www.postgresql.org/docs/11/sql-dropforeigndatawrapper.html) the [Foreign data wrapper](https://www.postgresql.org/docs/12/ddl-foreign-data.html)(beta feature).
> | yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (ysql_dumps).

Reviewer: ```suggestion
| Role | Description |
| :--- | :---------- |
| postgres | Superuser role used during database creation and for PostgreSQL operations. |
| yb_extension | Role that allows non-Superuser users to create PostgreSQL extensions. |
| yb_fdw | Role that allows non-Superuser users to [CREATE](https://www.postgresql.org/docs/11/sql-createforeigndatawrapper.html), [ALTER](https://www.postgresql.org/docs/11/sql-alterforeigndatawrapper.html), and [DROP](https://www.postgresql.org/docs/11/sql-dropforeigndatawrapper.html) the [Foreign data wrapper](https://www.postgresql.org/docs/12/ddl-foreign-data.html)(beta feature). |
| yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (ysql_dumps). |
```

### yb_extension

The `yb_extension` role allows non-Superuser roles to [create extensions](../../../api/ysql/the-sql-language/statements/ddl_create_extension/). A user granted this role can create all the extensions that are bundled in YugabyteDB.

Create a role `test` and grant `yb_extension` to this role.

```sql
yugabyte=# create role test;

> yugabyte=# grant yb_extension to test ;

Reviewer 2: remove spaces before semi colons; one below as well

* Roles inherit the privileges of all other roles granted to them.

> View the roles for your YugabyteDB clusters with the following command:

Reviewer: ```suggestion
YugabyteDB inherits a number of roles from PostgreSQL, including the `postgres` user, and adds several new roles. View the YugabyteDB-specific roles for your clusters with the following command (or use `\duS` to display all roles):
```

```sql
yugabyte=> \du
```

```output
                                     List of roles
  Role name   |                         Attributes                         | Member of
--------------+------------------------------------------------------------+-----------
 postgres     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
 yb_extension | Cannot login                                               | {}
 yb_fdw       | Cannot login                                               | {}
 yugabyte     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
```

The following table describes the default YSQL roles and users in YugabyteDB clusters.
<!-- Portions of this table are also under Database authorization in Yugabyte cloud -->

> | Role | Description |

Reviewer 3: So we have any more roles available in the cluster .  Use the command  select * from pg_roles;  to display all the roles. Is the intention to document all of those roles or  just which are added specifically by Yugabyte ?

Reviewer: @sushantrmishra I think it's sufficient to document just what we add, plus the `postgres` user (to show it's got the same attributes as `yugabyte`), so \du seems more useful here. I'll add a suggestion a few lines up to clarify the wording, though.

| :--- | :---------- |
| postgres | Superuser role created during database creation. |
| yb_extension | Role that allows non-Superuser users to create PostgreSQL extensions. |
| yb_fdw | Role that allows non-Superuser users to [CREATE](https://www.postgresql.org/docs/11/sql-createforeigndatawrapper.html), [ALTER](https://www.postgresql.org/docs/11/sql-alterforeigndatawrapper.html), and [DROP](https://www.postgresql.org/docs/11/sql-dropforeigndatawrapper.html) the [Foreign data wrapper](https://www.postgresql.org/docs/12/ddl-foreign-data.html)(beta feature). |

> | yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (ysql_dumps). |

Reviewer: ```suggestion
| yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (using ysql_dump). |
```

### yb_extension

> The `yb_extension` role allows non-Superuser roles to [create extensions](../../../api/ysql/the-sql-language/statements/ddl_create_extension/). A user granted this role can create all the extensions that are bundled in YugabyteDB.

Reviewer: ```suggestion
The `yb_extension` role allows non-superuser roles to [create extensions](../../../api/ysql/the-sql-language/statements/ddl_create_extension/). A user granted this role can create all the extensions that are bundled in YugabyteDB.
```

Create a role `test` and grant `yb_extension` to this role.

```sql
yugabyte=# create role test;
yugabyte=# grant yb_extension to test;
yugabyte=# set role test;
yugabyte=> select * from current_user;
```

```output
 current_user
--------------
 test
(1 row)
```

Create an extension as the test user and check if it's created.

```sql
yugabyte=> create extension pgcrypto;
yugabyte=> select * from pg_extension where extname='pgcrypto';
```

```output
 extname  | extowner | extnamespace | extrelocatable | extversion | extconfig | extcondition
----------+----------+--------------+----------------+------------+-----------+--------------
 pgcrypto |    16386 |         2200 | t              | 1.3        |           |
(1 row)
```

## Resources

> YSQL defines a number of specific resources, that represent underlying database objects. A resource can denote one object or a collection of objects. YSQL resources are hierarchical as described below:

Reviewer: ```suggestion
YSQL defines a number of specific resources that represent underlying database objects. A resource can represent one object or a collection of objects. YSQL resources are hierarchical as described below:
```
